torch_frame/data/multi_embedding_tensor.py

```diff
@@ -152,6 +285,60 @@ def _row_index_select(self, index: Tensor) -> 'MultiEmbeddingTensor':
             offset=self.offset,
         )
 
+    def _col_index_select(self, index: Tensor) -> 'MultiEmbeddingTensor':
+        if index.numel() == 0:
+            return MultiEmbeddingTensor(
+                num_rows=self.num_rows,
+                num_cols=0,
+                values=torch.tensor([], device=self.device),
+                offset=torch.tensor([0], device=self.device),
+            )
+        offset = torch.zeros(
+            index.size(0) + 1,
+            dtype=torch.long,
+            device=self.device,
+        )
+        col_dims = self.offset[1:] - self.offset[:-1]
+        new_col_dims = col_dims[index]
+        torch.cumsum(new_col_dims, dim=0, out=offset[1:])
+        value_index = [
+            torch.arange(
+                self.offset[i],
+                self.offset[i + 1],
+                dtype=torch.long,
+                device=self.device,
+            ) for i in index
+        ]
+        value_index = torch.cat(value_index, dim=0)
```

Comment:
Can you try using `batched_arange` to avoid for-loop? See my usage in `MultiNestedTensor`.